File: 350. Intersection_of_Two_Arrays_II/solution(improved).java
class Solution {
    public int[] intersect(int[] nums1, int[] nums2) {
        
        ArrayList<Integer> ans = new ArrayList<Integer>();
        
        int[] dict = new int[1001];
        Arrays.fill(dict, 0);
        
        for(int i = 0 ; i < nums1.length ; i++)
            dict[nums1[i]]++;
        
        for(int i = 0 ; i < nums2.length ; i++){
            if(dict[nums2[i]] > 0){
                dict[nums2[i]]--;
                ans.add(nums2[i]);
            }
        }
        
        int[] final_ans = new int[ans.size()];
        for(int i = 0 ; i < ans.size() ; i++)
            final_ans[i] = ans.get(i);
        return final_ans;
        
    }
}

File: 350. Intersection_of_Two_Arrays_II/solution(improved).cpp
class Solution {
public:
    vector<int> intersect(vector<int>& nums1, vector<int>& nums2) {
        
        vector<int> ans;
        
        int dict[1001];
        memset(dict, 0, sizeof(dict));
        
        for(int i = 0 ; i < nums1.size() ; i++)
            dict[nums1[i]]++;
        
        for(int i = 0 ; i < nums2.size() ; i++){
            if(dict[nums2[i]] > 0){
                dict[nums2[i]]--;
                ans.push_back(nums2[i]);
            }
        }
        
        return ans;
        
    }
};